cpp/zotaut/dump_delivered_marc_records.cc

```diff
@@ -79,19 +79,36 @@ void GetJournalEntriesFromDb(DbConnection * const db_connection, const std::stri
             const auto dois(record.getDOIs());
             csv_row += ";" + TextUtil::CSVEscape(StringUtil::Join(dois, '\n'));
 
-            for (const auto &_936_field : record.getTagRange("936")) {
-                csv_row += ";" + _936_field.getFirstSubfieldWithCode('j');
-                csv_row += ";" + _936_field.getFirstSubfieldWithCode('d');
-                csv_row += ";" + _936_field.getFirstSubfieldWithCode('e');
-                break;
+            const auto _773_field(record.findTag("773"));
+            const auto g_773_contents(_773_field->getFirstSubfieldWithCode('g'));
+            if (not g_773_contents.empty()) {
+                std::vector<std::string> subfields;
+                std::vector<std::string> filtered_dates;
+                for (const auto &field : record.getTagRange("773")) {
+                    if (field.getIndicator1() == '1') {
+                        for (const auto &subfield : field.getSubfields()) {
+                            StringUtil::Split(subfield.value_, ':', &subfields, true);
+                            filtered_dates.emplace_back(subfields[1]);
+                        }
+                    }
+                }
+                csv_row += ";" + filtered_dates[1];
+                csv_row += ";" + filtered_dates[0];
+                csv_row += ";" + filtered_dates[2];
+            } else {
+                for (const auto &_936_field : record.getTagRange("936")) {
```

Comment:
Also, please add correct indicator checks to all existing cases regarding 936 where they are missing